src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -202,17 +203,22 @@ private static Condition handleRegEx(RegexCondition regexCondition, EntityType e
 
   private static Condition handleContains(ContainsCondition containsCondition, EntityType entityType, Field<Object> field) {
     if (containsCondition.value() instanceof String value) {
-      return caseInsensitiveComparison(containsCondition, entityType, field, value, Field::containsIgnoreCase, Field::contains);
```

Comment:
I was originally trying to get the new logic working with the existing caseInsensitiveComparison method but it ended up becoming pretty messy due to the different data types. In the end I figured it would be easier to just cast the value to lowercase within the handleContains/handleNotContains methods themselves.